#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/lego/protos/LegoWheel.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.22724039092530346 0.8557920006949146 -0.4647384815987488 1.0381740912303312
  position -2.1883700837371007 2.785208505548386 5.737874332954219
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 2.79 1.73 0
  rotation 1 0 0 0
  floorSize 4 2.7
  floorAppearance Parquetry {
  }
}
Wall {
  translation 2.97028 1.7695 0
  rotation 0 0 1 1.57
  name "wall(19)"
  size 0.5 0.08 0.15
}
Wall {
  translation 3.23 1.551 0
  name "wall(18)"
  size 0.6 0.08 0.15
}
Wall {
  translation 3.25 2.531 0
  name "wall(17)"
  size 0.6 0.08 0.15
}
Wall {
  translation 3.5103 1.81886 0
  rotation 0 0 1 1.57
  name "wall(16)"
  size 1.5 0.08 0.15
}
Wall {
  translation 3.77 2.109 0
  name "wall(15)"
  size 0.6 0.08 0.15
}
Wall {
  translation 4.09943 0.88874 0
  rotation 0 0 1 1.57
  name "wall(14)"
  size 1 0.08 0.15
}
Wall {
  translation 4.10096 2.56862 0
  rotation 0 0 1 1.57
  name "wall(13)"
  size 1 0.08 0.15
}
Wall {
  translation 2.71935 0.68978 0
  rotation 0 0 1 1.57
  name "wall(12)"
  size 0.6 0.08 0.15
}
Wall {
  translation 2.46 0.951 0
  name "wall(11)"
  size 0.6 0.08 0.15
}
Wall {
  translation 2.08 1.421 0
  name "wall(9)"
  size 0.7 0.08 0.15
}
Wall {
  translation 1.77 2.36081 0
  rotation 0 0 1 1.57
  name "wall(7)"
  size 0.6 0.08 0.15
}
Wall {
  translation 1.51 2.101 0
  name "wall(6)"
  size 0.6 0.08 0.15
}
Wall {
  translation 1.25095 2.56102 0
  rotation 0 0 1 1.57
  name "wall(4)"
  size 1 0.08 0.15
}
Wall {
  translation 2.39046 1.97994 0
  rotation 0 0 1 1.57
  name "wall(10)"
  size 1.2 0.08 0.15
}
Wall {
  translation 1.76956 0.960604 0
  rotation 0 0 1 1.57
  name "wall(3)"
  size 1 0.08 0.15
}
Wall {
  translation 1.24956 0.96102 0
  rotation 0 0 1 1.57
  name "wall(2)"
  size 1 0.08 0.15
}
Wall {
  translation 2.674 3.03 0
  size 2.93 0.08 0.15
}
Wall {
  translation 2.674 0.43 0
  name "wall(1)"
  size 2.93 0.08 0.15
}
Robot {
  translation 2.410531026420486 1.3032906926988261 0.07189478780046217
  rotation 6.453717683938913e-05 -0.00033699296907393025 -0.9999999411353442 2.8874688605467043
  children [
    DEF Sensor1 DistanceSensor {
      translation 0.05 0 0.01
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor1"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor2 DistanceSensor {
      translation 0.05 0.04 0.01
      rotation 0 0 1 0.7853996938995746
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor2"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor3 DistanceSensor {
      translation 0 0.04 0.01
      rotation 0 0 1 1.57079633
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor3"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor4 DistanceSensor {
      translation -0.05 0.04 0.01
      rotation 0 0 1 2.35619449
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor4"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor5 DistanceSensor {
      translation -0.05 0 0.01
      rotation 0 0 1 3.14159265
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor5"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor6 DistanceSensor {
      translation -0.05 -0.04 0.01
      rotation 0 0 1 -2.3561944871795863
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor6"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor7 DistanceSensor {
      translation 0 -0.04 0.01
      rotation 0 0 1 -1.5707963271795862
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor7"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Sensor8 DistanceSensor {
      translation 0.05 -0.04 0.01
      rotation 0 0 1 -0.7853996938995746
      children [
        DEF Sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      name "Sensor8"
      boundingObject USE Sensor
      physics Physics {
      }
    }
    DEF Right2 HingeJoint {
      jointParameters HingeJointParameters {
        position 60391.600691581
        axis 0 1 0
        anchor 0.03 -0.05 -0.05
      }
      device [
        RotationalMotor {
          name "Right2"
        }
      ]
      endPoint LegoWheel {
        hidden linearVelocity_0 -0.06674358577644704 -0.018300113297033113 1.065278538101713e-05
        hidden angularVelocity_0 0.7381854461333419 -3.059220516504725 0.6518414991167928
        translation 0.030002008741859418 -0.050000000400255304 -0.05000159024076607
        rotation -0.006214389363832518 -0.999963153934604 0.005922173418117043 2.3765983593882067
        name "Right2"
      }
    }
    DEF Left2 HingeJoint {
      jointParameters HingeJointParameters {
        position 100630.11745863488
        axis 0 1 0
        anchor 0.03 0.05 -0.05
      }
      device [
        RotationalMotor {
          name "Left2"
        }
      ]
      endPoint LegoWheel {
        hidden linearVelocity_0 -0.003746123371288726 -0.0025212048996972464 -0.000698178969793182
        hidden angularVelocity_0 -1.5052960029559883 6.089953382262278 0.6306098126635609
        translation 0.030002506523273548 0.049999999608291545 -0.0499999461785252
        rotation -0.011843125256575288 0.9999226981593403 -0.003786567033875326 4.904692606389555
        name "Left2"
      }
    }
    DEF Right1 HingeJoint {
      jointParameters HingeJointParameters {
        position 60389.48433271031
        axis 0 1 0
        anchor -0.03 -0.05 -0.05
      }
      device [
        RotationalMotor {
          name "Right1"
        }
      ]
      endPoint LegoWheel {
        hidden linearVelocity_0 -0.0762009154348077 0.019423631917708926 0.0004321999748168157
        hidden angularVelocity_0 0.7381845097823906 -3.059220916829201 0.6518414244083713
        translation -0.029998049454937604 -0.05000000038863325 -0.05000154416728445
        rotation 0.00250656624520962 -0.9999481879058828 0.009866034127229388 4.492881122816066
        name "Right1"
      }
    }
    DEF Left1 HingeJoint {
      jointParameters HingeJointParameters {
        position 100630.11944909772
        axis 0 1 0
        anchor -0.03 0.05 -0.05
      }
      device [
        RotationalMotor {
          name "Left1"
        }
      ]
      endPoint LegoWheel {
        hidden linearVelocity_0 -0.013214968670970967 0.03521490078093417 -0.00027843835913260115
        hidden angularVelocity_0 -1.5052968093772086 6.089953339163394 0.6306097635623568
        translation -0.029997459071454177 0.049999999602955675 -0.0499999454448672
        rotation -0.011853328368591548 0.9999225129485412 -0.0038035122270010297 4.9066829157180205
        name "Left1"
      }
    }
    DEF RobotBody Shape {
      appearance PBRAppearance {
        baseColor 1 0.4 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.08 0.08
      }
    }
  ]
  boundingObject USE RobotBody
  physics Physics {
  }
  controller "RWallFollow"
  linearVelocity -0.0402612014648098 0.008006248735935568 -0.00013768007760319812
  angularVelocity 0.008804551793685032 -0.005107486902082327 0.6493295996738058
}
